compiler/rustc_middle/src/ty/predicate.rs

```diff
@@ -120,6 +116,8 @@ impl<'tcx> Predicate<'tcx> {
         Some(tcx.mk_predicate(kind))
     }
 
+    /// Only used by the old solver to decide whether a predicate is accepted
```

Comment:
We could just remove this by inlining this into the old solver's (one?) use-site. Could you add a FIXME?